File: 정렬/H_index.java
package sorting;

import java.util.Arrays;

public class H_index {
	 public int solution(int[] citations) {
	        int answer = 0;
	        
	        Arrays.sort(citations);
	        int le = citations.length;
	        
	        for(int i=0;i<le;i++){
	            if((((le-1)-i)+1)<=citations[i]){
	                answer=((le-1)-i)+1; //citations[i]가 되지 않는 이유는?
	                break;}
	        }
	        
	        
	        return answer;
	    }
}

File: 정렬/hindex.py
def solution(citations):
    citations.sort()  # 정렬을 해놓는 것이 중요. 정렬 이후엔 해당 원소 뒤에는 다 그것보다 많이 인용된 것.

    l = len(citations)
    for i in range(l):

        if citations[i] >= l - i:
            # 논문이 인용된 횟수(h번 이상) >= 인용된 논문의 개수(h개 == h번)
            return l - i
    return 0